### PR TITLE
Changes reload/start/restart.

### PR DESCRIPTION
**Pull request by damencho**
When reloading we want to restart the process when idle, so we exit with
 status 0 and systemd will restart the process.
When stopping graceful or not we want to exit the process with status as
 configured in systemd unit-file in RestartPreventExitStatus=255.

### CODE REVIEW

File: src/main/kotlin/org/jitsi/jibri/Main.kt [hunk 1]
```diff
@@ -125,14 +125,24 @@ fun main(args: Array<String>) {
             exitProcess(0)
         }
     }
+    val gracefulShutdownHandler = {
+        logger.info("Jibri has been told to graceful shutdown, waiting to be idle before exiting")
+        jibri.executeWhenIdle {
+            logger.info("Jibri is idle and there are config file changes, exiting")
```
Comment (bbaldino) [inline src/main/kotlin/org/jitsi/jibri/Main.kt]: `"Jibri is idle and has been told to gracefully shutdown, exiting")`

File: src/main/kotlin/org/jitsi/jibri/Main.kt [hunk 2]
```diff
@@ -125,14 +125,24 @@ fun main(args: Array<String>) {
...
+            // Exit with code 255 to indicate we do not want process restart
+            logger.info("Service stopped")
```
Comment (bbaldino) [inline src/main/kotlin/org/jitsi/jibri/Main.kt]: don't think this log is needed?

File: src/test/kotlin/org/jitsi/jibri/api/http/internal/InternalHttpApiTest.kt [hunk 3]
```diff
@@ -46,20 +46,33 @@ class InternalHttpApiTest : ShouldSpec() {
...
+        "notifyConfigChanged" {
+            should("return a 200 and not invoke the config changed handler directly") {
+                var configChangedHandlerCalled = false
+                val configChangedHandler = {
+                    configChangedHandlerCalled = true
+                }
+                val internalHttpApi = InternalHttpApi({}, configChangedHandler, {})
+                val response = internalHttpApi.reloadConfig()
+                response.status shouldBe 200
+                configChangedHandlerCalled shouldBe false
```
Comment (bbaldino) [inline src/test/kotlin/org/jitsi/jibri/api/http/internal/InternalHttpApiTest.kt]: realizing now that these tests should really verify that the handler does _eventually_ get called...but i'll go in and fix that later

File: src/main/kotlin/org/jitsi/jibri/Main.kt (hunk 1 above)
Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/Main.kt]: ?

Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/Main.kt]: Got it

File: src/main/kotlin/org/jitsi/jibri/Main.kt (hunk 2 above)
Comment (damencho) [reply src/main/kotlin/org/jitsi/jibri/Main.kt]: removed

**Comment by bbaldino** [comment]
lgtm, but looks like you'll need to rebase to resolve some conflicts

**Comment by damencho** [comment]
Rebased, should be fine now.

File: resources/debian-package/opt/jitsi/jibri/reload.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+curl -X POST http://127.0.0.1:3333/jibri/api/internal/v1.0/notifyConfigChanged
```
Comment (saghul) [inline resources/debian-package/opt/jitsi/jibri/reload.sh]: Can't we use SIGHUP instead? It's a more common approach to reloading the config of a running process.

Comment (damencho) [reply resources/debian-package/opt/jitsi/jibri/reload.sh]: I think there is no general mechanism in java to listen for those ...

Comment (saghul) [reply resources/debian-package/opt/jitsi/jibri/reload.sh]: ` For the signals SIGINT, SIGTERM, and SIGHUP the Java Runtime is terminated by invoking the equivalent of System.exit with a value of 128 + the signal number. Other signals are ignored. `

Ouch. Ignore what I said...

**Pull request merged**